src/components/molecules/VenueJoinButton/VenueJoinButton.tsx

```diff
@@ -0,0 +1,180 @@
+  venueId?: string;
+  venue: AnyVenue;
+  onPasswordSubmit?: () => void;
+}
+
+export interface VenueJoinButtonData {
+  password: string;
+}
+
+export const VenueJoinButton: React.FC<VenueJoinButtonProps> = ({
+  venueId,
+  venue,
+  onPasswordSubmit,
+}) => {
+  const { user } = useUser();
+  const [message, setMessage] = useState("");
+  const [isJoinVenueModalOpen, setJoinVenueModalOpen] = useState(false);
+
+  const isLoggedIn = !!user;
+  const buttonText =
+    venue.config?.landingPageConfig.joinButtonText ?? DEFAULT_PARTY_BUTTON_TEXT;
+
+  const {
+    register,
+    handleSubmit,
+    errors,
+    setError,
+    clearError,
+  } = useForm<VenueJoinButtonData>({
+    mode: "onChange",
+    reValidateMode: "onChange",
+  });
+
+  const showJoinVenueModal = useCallback(() => {
+    setJoinVenueModalOpen(true);
+  }, []);
+
+  const hideJoinVenueModal = useCallback(() => {
+    setJoinVenueModalOpen(false);
+  }, []);
+
+  const handleJoinVenue = useCallback(() => {
+    if (!venueId) return;
+
+    joinVenue(venueId, { hasVenueEntrance: !!venue.entrance?.length });
+  }, [venue, venueId]);
+
+  const changePassword = useCallback(() => {
+    setMessage("");
+    clearError();
+  }, [clearError]);
+
+  const submitPassword = useCallback(
+    async ({ password }) => {
+      if (!venueId) return;
+
+      if (!venue.access) {
+        handleJoinVenue();
+        return;
+      }
+
+      setMessage("Checking password...");
+      onPasswordSubmit && onPasswordSubmit();
+
+      if (!user) {
+        setMessage("");
+        return;
+      }
+
+      await checkAccess({
+        venueId,
+        password,
+      })
+        .then((result) => {
+          if (isTruthy(result?.data?.token)) {
+            setLocalStorageToken(venueId, result.data.token);
+            showJoinVenueModal();
+            setMessage("Success!");
+          } else {
+            setMessage("");
+            setError("password", "required", `Wrong password!`);
+          }
+        })
+        .catch((error) => {
+          setMessage(`Password error: ${error.toString()}`);
+        });
+    },
+    [
+      handleJoinVenue,
+      onPasswordSubmit,
+      setError,
+      showJoinVenueModal,
+      user,
+      venue.access,
+      venueId,
+    ]
+  );
+
+  return (
+    <>
+      <div className="secret-password-form-wrapper">
+        <form
+          className="secret-password-form"
+          onSubmit={handleSubmit(submitPassword)}
+        >
+          {venue.access && (
+            <>
+              <p className="small-text">
+                Got an invite? Join in with the secret password
+              </p>
+              <input
+                className="secret-password-input"
+                name="password"
+                ref={register({ required: true })}
+                placeholder="password"
+                autoFocus
+                onChange={changePassword}
+                id="password"
+              />
+            </>
+          )}
```

Comment:
Presumably this is going to need to change to support other forms of venue access? Should we be checking `venue.access === 'foo-password-mode-whatever-it-is'` here?